### PR TITLE
Make "-open" command line flag accept a module path

### PR DESCRIPTION
**Pull request by aalekseyev**
This is a rebased patch from Mantis issue [MPR#7137](http://caml.inria.fr/mantis/view.php?id=7137).

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -100,6 +100,9 @@ Next version (4.05.0):
 - clarify ocamldoc text parsing error messages
   (Gabriel Scherer)
 
+- Make "-open" command line flag accept a module path (not a module name)
```
Comment (gasche) [inline Changes]: Giving the issue numbers is important for traceability. Your entry should start with `PR#7137, GPR#960`, and given the entry ordering (by MPR first, then by GPR), it should be the first in the section.

Comment (aalekseyev) [reply Changes]: I added these issue numbers, improved wording and moved the changelog entry to a section that seems more appropriate.

**Comment by gasche** [comment]
The changelog entry is still at the wrong place ordering-wise in its new section, but oh well. I'll merge now.

**Comment by damiendoligez** [comment]
The test is wrong:
* "make clean" fails when the directory is already clean
* "make" doesn't return any test results

**Comment by gasche** [comment]
Sorry about that, I will rewrite the test.

**Comment by aalekseyev** [comment]
Sorry about that. Thank you @gasche!

**Comment by Octachron** [comment]
The ocamldoc side has not been updated (see https://github.com/ocaml/ocaml/blob/trunk/ocamldoc/odoc_analyse.ml#L53). Should I fix this point?

**Comment by aalekseyev** [comment]
Sigh, I should have tried harder to change the type of open_modules.

Yes please.

**Pull request merged**